benchmarks/dataset_serialize_benchmark.py

```diff
@@ -0,0 +1,232 @@
+            format=source.format_str,
+        )
+
+    def run(self, source, case=None, **kwargs):
+
+        cases = self.get_cases(case, kwargs)
+
+        for source in self.get_sources(source):
+
+            log.info("source %s: download, if required", source.name)
+            source.download_source_if_not_exists()
+            tags = self.get_tags(kwargs, source)
+
+            t0 = time.monotonic()
+            source_ds = self._get_dataset_for_source(source)
+            log.info(
+                "constructed Dataset object for source in %.4f s", time.monotonic() - t0
+            )
+
+            for case in cases:
+
+                log.info("case %s: create directory", case)
+                dirpath = self._create_tmpdir_in_ramdisk(case)
+                log.info("directory created, path: %s", dirpath)
+
+                yield self.benchmark(
+                    f=self._get_benchmark_function(
+                        case, source.name, source_ds, dirpath
+                    ),
+                    extra_tags=tags,
+                    options=kwargs,
+                    case=case,
+                )
+
+                # Free up memory in the RAM disk (tmpfs), assuming that we're
+                # otherwise getting close to filling it (depending on the
+                # machine this is executed on, a single test might easily
+                # occupy 10 % or more of this tmpfs).
+                log.info("removing directory: %s", dirpath)
+                shutil.rmtree(dirpath)
+
+    def _get_benchmark_function(
+        self, case, source_name: str, source_ds: ds.Dataset, dirpath: str
+    ):
+
+        (selectivity, serialization_format) = case
+
+        # Option A: read-from-disk -> deserialize -> filter -> into memory
+        # before timing serialize -> write-to-tmpfs
+        t0 = time.monotonic()
+        data = source_ds.to_table(filter=self.filters[source_name][selectivity])
+        log.info("read source dataset into memory in %.4f s", time.monotonic() - t0)
+
+        # Option B (thrown away, but kept for posterity): use a Scanner()
+        # object to transparently filter the source dataset upon consumption,
+        # in which case what's timed is read-from-disk -> deserialize -> filter
+        # -> serialize -> write-to-tmpfs
+        #
+        # Note(JP): I have confirmed that for the data used in this benchmark
+        # this option is dominated by read-from-disk to the extent that no
+        # useful signal is generated for the write phase, at least for some
+        # serialization formats.
+        #
+        # data = pyarrow.dataset.Scanner.from_dataset(source_ds,
+        #     filter=self.filters[source_name][selectivity])
+
+        def benchfunc():
+            # This is a hack to make each iteration work in a separate
+            # directory (otherwise some write operations would error out saying
+            # that the target directory is not empty). With `benchrun` it will
```

Comment:
If you don't like this, there is an `existing_data_behavior` parameter in `write_dataset` you could use

Reply:
Oh, cool. Thanks for pointing that out! Documented here: 

https://arrow.apache.org/docs/python/generated/pyarrow.dataset.write_dataset.html

```
existing_data_behavior‘error’ | ‘overwrite_or_ignore’ | ‘delete_matching’
```

The current approach creates a fresh directory for each iteration and should therefore be 'fastest'. Deletion might take valuable time (which would be timed).

It _might be_ that `‘overwrite_or_ignore’` is actually the smartest _if_ the second+ write consumes as much time as the first one. This would be the best option to limit space consumption across iterations.